### PR TITLE
chore: gateway cleanup

### PR DESCRIPTION
**Pull request by tvolk131**
A bunch of small tweaks and cleanups throughout the gateway code. I recommend reviewing commit-by-commit.

### CODE REVIEW

File: gateway/ln-gateway/src/lib.rs [hunk 1]
```diff
@@ -486,6 +488,22 @@ impl Gateway {
...
+
+    pub fn versioned_api(&self) -> &SafeUrl {
+        &self.versioned_api
+    }
+
+    pub async fn clone_gateway_config(&self) -> Option<GatewayConfiguration> {
+        self.gateway_config.read().await.clone()
+    }
+
+    pub async fn clone_state(&self) -> GatewayState {
```
Comment (dpc) [inline gateway/ln-gateway/src/lib.rs:503]: I kind of don't see a point of `clone_` here. This type seems so cheap, that we might just call it `get_state`.

**Review by dpc — APPROVED** [review]
Haven't noticed anything controversial.

**Review by m1sterc001guy — DISMISSED** [review]
LGTM, just needs rebase

File: gateway/ln-gateway/src/lib.rs (hunk 1 above)
Comment (tvolk131) [reply gateway/ln-gateway/src/lib.rs]: Done.

**Comment by tvolk131** [comment]
> just needs rebase

Rebased!

**Review by m1sterc001guy — APPROVED** [review]

**Pull request merged**